blocks/inspector-controls/select-control/index.js

```diff
@@ -14,23 +14,22 @@ import { withInstanceId } from 'components';
 import BaseControl from './../base-control';
 import './style.scss';
 
-function SelectControl( { label, selected, instanceId, onBlur, options = [], ...props } ) {
+function SelectControl( { label, selected, instanceId, onChange, options = [], ...props } ) {
 	const id = 'inspector-select-control-' + instanceId;
-	const onBlurValue = ( event ) => onBlur( event.target.value );
+	const onChangeValue = ( event ) => onChange( event.target.value );
 
 	return ! isEmpty( options ) && (
 		<BaseControl label={ label } id={ id }>
 			<select
 				id={ id }
 				className="blocks-select-control__input"
-				onBlur={ onBlurValue }
+				onChange={ onChangeValue }
```

Comment:
While we chose `onBlur` in the case of the Mode Switcher, we still opt in to the rule and ideas behind `jsx-a11y/no-onchange`:

https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-onchange.md

There's some related ongoing discussion at #392.

I'm not sure we'd want to make the move from `onBlur` to `onChange` in this case.